### PR TITLE
feat(your-work): GitHub repo filter

### PR DESCRIPTION
**Pull request by jmtaber129**
# Description

Fixes https://github.com/ParabolInc/parabol/issues/8723

Add a repo filter to the GitHub panel of the Your Work drawer.

Also refactor out the GH repo filter from Sprint Poker for reuse.

## Demo
https://www.loom.com/share/9a8903afac134c2abfeff2fc6fb9cbfa?sid=42540c12-609a-4116-aef1-1f5933c9bfc7

Screenshots with filter icon:
Screen Shot 2023-10-05 at 2 46 36 PM
Screen Shot 2023-10-05 at 2 46 40 PM

## Testing scenarios
- [ ] Test the filter for one or more repositories
- [ ] Smoke test the repo filter in Sprint Poker

## Final checklist

- [X] I checked the [code review guidelines](../docs/codeReview.md)
- [X] I have added [Metrics Representative](../docs/codeReview.md#metrics-representative) as reviewer(s) if my PR invovles metrics/data/analytics related changes
- [X] I have performed a self-review of my code, the same way I'd do it for any other team member
- [X] I have tested all cases I listed in the testing scenarios and I haven't found any issues or regressions
- [X] Whenever I took a non-obvious choice I added a comment explaining why I did it this way
- [X] I added the label `One Review Required` if the PR introduces only minor changes, does not contain any architectural changes or does not introduce any new patterns and I think one review is sufficient'
- [X] PR title is human readable and could be used in changelog

### CODE REVIEW

File: packages/client/components/TeamPrompt/WorkDrawer/GitHubIntegrationResults.tsx [hunk 1]
```diff
@@ -64,6 +64,7 @@ const GitHubIntegrationResults = (props: Props) => {
                 }
               }
             }
+            ...useGetRepoContributions_teamMember
```
Comment (Dschoordsch) [inline packages/client/components/TeamPrompt/WorkDrawer/GitHubIntegrationResults.tsx:67]: -1 This seems wrong, we're not using this hook in this component, why do we need to include this fragment then?

**Review by Dschoordsch — CHANGES_REQUESTED** [review]
Looks good, only small nits

File: static/images/icons/repos.svg
[the remark this answers is not in the document]
Comment (jmtaber129) [reply static/images/icons/repos.svg:1]: @enriquesanchez thoughts on this?

[the remark this answers is not in the document]
Comment (enriquesanchez) [reply static/images/icons/repos.svg:1]: @jmtaber129 Oh! Georg is correct, we should use the filter icon instead. Thank you @Dschoordsch 🙌

[the remark this answers is not in the document]
Comment (jmtaber129) [reply static/images/icons/repos.svg:1]: Updated!

File: packages/client/components/TeamPrompt/WorkDrawer/GitHubIntegrationResults.tsx (hunk 1 above)
Comment (jmtaber129) [reply packages/client/components/TeamPrompt/WorkDrawer/GitHubIntegrationResults.tsx]: Removed

**Review by Dschoordsch — APPROVED** [review]

**Comment by Dschoordsch** [comment]
Merging it because James is out today and Monday

**Pull request merged**